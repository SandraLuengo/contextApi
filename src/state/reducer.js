import chinesCities from '../fixtures/cities-of-china.json';

const initialState = {
  cities: chinesCities.cities.map(city => ({ ...city, isSelected: false })),
  searching: '',
  isAllChecked: false,
};


const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_CITIES':
      return {
        ...state,
        cities: action.newValue,
      };
    case 'SET_SEARCH':
      return {
        ...state,
        searching: action.newValue,
      };
    case 'SET_IS_ALL_CHECKED':
      return {
        ...state,
        isAllChecked: action.newValue,
      };
    default:
      return state;
  }
};

export { initialState, reducer };

