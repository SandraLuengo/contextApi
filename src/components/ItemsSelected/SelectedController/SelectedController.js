import React from 'react';
import { useGlobalState } from '../../../state/useGlobalState';
import { clearAll } from '../../functions';

const SelectedController = ({ className }) => {
  const [{ cities }, dispatch] = useGlobalState();

  const selected = () => cities.filter(city => city.isSelected).length;

  return (
    <div className={className}>
      <p className="itemsNumber">{selected()} items</p>
      <p
        className="clear"
        onClick={() => {
          clearAll(cities, dispatch);
          dispatch({ type: 'SET_IS_ALL_CHECKED', newValue: false });
        }}
      >
        CLEAR
      </p>
    </div>
  );
};

export default SelectedController;
