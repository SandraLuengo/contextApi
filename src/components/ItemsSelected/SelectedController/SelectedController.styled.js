import styled from 'styled-components';
import SelectedController from './SelectedController';
import * as style from '../../../static/styles/stylesVariables';

export default styled(SelectedController).attrs({})`
  max-width: 100%;
  height: 8vh;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-style: none none solid none;
  border-color: #e9e9e9;
  border-width: 1.3px;
  padding: 0 3%;
  .itemsNumber {
    color: #979797;
  }
  .clear {
    color: ${style.primaryColor};
    font-weight: bold;
    &:hover {
      cursor: pointer;
      color: #4d8dd6;
      transition: color .3s ease-in-out;
    }
  }
`;
