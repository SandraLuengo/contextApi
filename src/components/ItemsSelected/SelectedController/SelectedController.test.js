import React from 'react';
import renderer from 'react-test-renderer';
import SelectedController from './SelectedController';
import { Provider } from '../../../state/useGlobalState';
import { initialState, reducer } from '../../../state/reducer';

const props = { className: 'test' };

it('SelectedController component renders correctly', () => {
  const tree = renderer
    .create(<Provider reducer={reducer} initialState={initialState}>
      <SelectedController {...props} />
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
