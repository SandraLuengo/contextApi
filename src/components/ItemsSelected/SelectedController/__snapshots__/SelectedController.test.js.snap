// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedController component renders correctly 1`] = `
<div
  className="test"
>
  <p
    className="itemsNumber"
  >
    0
     items
  </p>
  <p
    className="clear"
    onClick={[Function]}
  >
    CLEAR
  </p>
</div>
`;
